### PR TITLE
Update documentation to RFC9562

### PR DESCRIPTION
**Pull request by pmccarren**
This PR updates documentation to reflect [RFC4122](https://datatracker.ietf.org/doc/html/rfc4122) becoming obsolete with the introduction of [RFC9562](https://datatracker.ietf.org/doc/html/rfc9562).

To maintain discoverability, the main top-level README maintains a reference to 4122. Additionally, `package.json` maintains the `rfc4122` keyword while adding the `rfc9562`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,12 +2,11 @@
...
   -->
 
-
 # uuid [![CI](https://github.com/uuidjs/uuid/workflows/CI/badge.svg)](https://github.com/uuidjs/uuid/actions?query=workflow%3ACI) [![Browser](https://github.com/uuidjs/uuid/workflows/Browser/badge.svg)](https://github.com/uuidjs/uuid/actions?query=workflow%3ABrowser)
 
-For the creation of [RFC4122](https://www.ietf.org/rfc/rfc4122.txt) UUIDs
+For the creation of [RFC9562](https://www.rfc-editor.org/rfc/rfc9562.html) (formally [RFC4122](https://www.rfc-editor.org/rfc/rfc4122.html)) UUIDs
 
-- **Complete** - Support for RFC4122 version 1, 3, 4, 5, and 7 UUIDs
+- **Complete** - Support for RFC9562 version 1, 3, 4, 5, and 7 UUIDs
```
Comment (broofa) [inline README.md:9]: Let's just commit to supporting all official RFC versions (1,3-8).  I'm working on v6 now, and v8 is trivial (just need to add support in `validate()` and `version()`)
```suggestion
- **Complete** - Support for all RFC9562  UUID versions
```

File: package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "uuid",
   "version": "9.0.1",
-  "description": "RFC4122 (v1, v3, v4, v5, and v7) UUIDs",
+  "description": "RFC9562 (v1, v3, v4, v5, and v7) UUIDs",
```
Comment (broofa) [inline package.json:4]:
```suggestion
  "description": "RFC9562 UUIDs",
```

**Review by broofa — APPROVED** [review]
Couple of nits, otherwise looks good.

File: README.md (hunk 1 above)
Comment (pmccarren) [reply README.md]: Great idea. I'll take v8 while you're working on v6

Comment (broofa) [reply README.md]: Sorry, 'didn't see this comment until just now.   I appreciate you creating the issue to kick off this work, though.  I should have done that.  My bad!

**Pull request merged**